test/lint/lxd-agent-imports.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -eu
+
+echo "Checking for imports that have been added to the lxd-agent..."
+
+OUT=$(go list -f '{{ join .Imports "\n" }}' ./lxd-agent | grep -F . | sort -u | diff -u test/lxd-agent-godeps.list - || true)
```

Comment:
@tomponline is it enough to check `./lxd-agent` alone here?

Reply:
This seems to only track the packages explicitly imported, and not any additional chained dependencies (which was the problem we saw before).

It looks like we can do `go list -f '{{ join .Deps "\n" }}' ./lxd-agent` to get the full list.

And I wonder if we shouldn't also be doing that for `./client` and `./lxc` separately (rather than combined as it is now). Then we wouldn't need to also check `./lxc/config` as well.

Reply:
Good idea. Would you like me to put them all in a single script each comparing against a `godeps.list` file? Or you'd prefer 3 scripts (`client`, `lxc` & `lxd-agent`)?

In any case, I'm not sure what to do of `./shared/api` if you want me to rework `client-imports.sh`.

Reply:
I don't mind how many scripts - but I think we should have a godeps.list file for each (client, lxc, lxd-agent and shared/api).